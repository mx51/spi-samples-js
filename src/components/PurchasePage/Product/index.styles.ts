import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';

export default makeStyles((theme: Theme) =>
  createStyles({
    cardStyle: {
      width: '12.625rem',
      height: '12.813rem',
      borderRadius: 4,
      margin: theme.spacing(1),
      border: 0,
    },
    productText: {
      height: '8.938rem',
    },
    productName: {
      maxHeight: '3.875rem',
      maxWidth: '12.625rem',
      borderRadius: 0,
      fontSize: '1rem',
      fontWeight: 500,
    },
    productImage: { width: '12.625rem', height: '8.938rem' },
  })
);
