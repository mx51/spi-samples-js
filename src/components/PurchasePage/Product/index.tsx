import React from 'react';
import { Box, Card, CardActionArea, CardContent, CardMedia, Typography } from '@material-ui/core';
import useStyles from './index.styles';
import { ProductProps } from './interfaces';

export default function Product({ product }: ProductProps): React.ReactElement {
  const classes = useStyles();
  return (
    <Box>
      <Card className={classes.cardStyle}>
        <CardActionArea>
          <CardMedia className={classes.productText} title={product.name}>
            <Box
              className={classes.productImage}
              bgcolor={product.backgroundColor}
              display="flex"
              justifyContent="center"
            >
              {product.image}
            </Box>
          </CardMedia>
          <CardContent>
            <Typography className={classes.productName} variant="h5" component="h2">
              {product.name}
            </Typography>
            <Typography variant="body2" color="textSecondary" component="p">
              {product.price}
            </Typography>
          </CardContent>
        </CardActionArea>
      </Card>
    </Box>
  );
}
