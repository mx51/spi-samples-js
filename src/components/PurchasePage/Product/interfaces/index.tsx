import { IProduct } from '../../ProductList/interfaces';

export type ProductProps = {
  product: IProduct;
};
