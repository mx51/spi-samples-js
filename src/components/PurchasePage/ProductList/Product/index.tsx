import React from 'react';
import { Box, Card, CardActionArea, CardContent, CardMedia, Typography } from '@material-ui/core';
import useStyles from './index.styles';
import { ProductProps } from './interfaces';

export default function Product({ product }: ProductProps): React.ReactElement {
  const classes = useStyles();
  return (
    <Box>
      <Card className={classes.cardStyle}>
        <CardActionArea>
          <CardMedia className={classes.productText} title={product.name}>
            <Box className={classes.productImage} display="flex" justifyContent="center">
              <img src={product.image} alt={product.name} />
            </Box>
          </CardMedia>
          <CardContent>
            <Typography className={classes.productName} variant="h5" component="h2">
              {product.name}
            </Typography>
            <Typography variant="body2" color="textSecondary" component="p">
              {Intl.NumberFormat('en-AU', { style: 'currency', currency: 'AUD' }).format(product.price)}
            </Typography>
          </CardContent>
        </CardActionArea>
      </Card>
    </Box>
  );
}
