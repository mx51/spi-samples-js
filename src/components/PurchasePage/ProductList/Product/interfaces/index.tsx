import { IProduct } from '../../interfaces';

export type ProductProps = {
  product: IProduct;
};
