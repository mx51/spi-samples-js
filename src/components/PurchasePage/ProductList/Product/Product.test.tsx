import React from 'react';
import { render } from '@testing-library/react';

import '@testing-library/jest-dom/extend-expect';
import Product from '.';

describe('Test <Product />', () => {
  test('snapshot for Product component', () => {
    const product = {
      name: 'Mocha',
      price: 420,
      image: 'images/product/mocha.png',
    };
    const onClick = jest.fn();

    const { container } = render(<Product product={product} onClick={onClick} />);
    expect(container).toMatchSnapshot();
  });
});
