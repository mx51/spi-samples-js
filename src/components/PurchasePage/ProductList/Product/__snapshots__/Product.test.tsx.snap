// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Product /> snapshot for Product component 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-5"
  >
    <div
      class="MuiPaper-root MuiCard-root makeStyles-cardStyle-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardMedia-root makeStyles-productText-2"
          title="Mocha"
        >
          <div
            class="MuiBox-root MuiBox-root-6 makeStyles-productImage-4"
          >
            <img
              alt="Mocha"
              src="images/product/mocha.png"
            />
          </div>
        </div>
        <div
          class="MuiCardContent-root"
        >
          <h2
            class="MuiTypography-root makeStyles-productName-3 MuiTypography-h5"
          >
            Mocha
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            $4.20
          </p>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
